src/lib/DatePickerComponent.android.js

```diff
@@ -2,153 +2,202 @@
 
-    const dateToSet = formatDateResult(date, mode);
+        onChange(value);
+      }
+      if (onValueChange) {
+        onValueChange(dateToSet);
+      }
+    } else {
+      this.setState({ date: "" });
 
-    this.setState(
-      {
-        date: dateToSet,
-        isDatePickerVisible: false
-      },
-      () => {
-        if (mode === "datetime") {
-          this.setState({
-            isTimePickerVisible: true
-          });
-        }
+      if (onChange) {
+        onChange(date);
       }
-    );
 
-    onChange &&
-      onChange(prettyPrint ? dateTimeFormat(dateToSet, mode) : dateToSet);
-    onValueChange && onValueChange(dateToSet);
+      if (onValueChange) {
+        onValueChange(date);
+      }
+    }
   }
 
   setDate(date) {
     const { mode } = this.props;
     const dateToSet = formatDateResult(date, mode);
     this.setState({ date: dateToSet });
 
-    if (this.props.onChange)
-      this.props.onChange(
-        this.props.prettyPrint
-          ? this.props.dateTimeFormat(dateToSet)
-          : dateToSet
-      );
-    if (this.props.onValueChange) this.props.onValueChange(dateToSet);
+    handleSetDate(dateToSet, this.props);
   }
 
   async _togglePicker(event) {
-    if (this.props.mode === "time") {
-      this.setState({ isTimePickerVisible: true });
-    } else {
-      this.setState({ isDatePickerVisible: true });
+    const key =
+      this.props.mode === "time"
+        ? "isTimePickerVisible"
+        : "isDatePickerVisible";
+
+    this.setState({ [key]: true });
+
+    if (this.props.onPress) {
+      this.props.onPress(event);
     }
+  }
 
-    this.props.onPress && this.props.onPress(event);
+  handleClear(e) {
+    const { mode } = this.props;
+
+    if (mode === "time") {
+      this.handleTimeValueChange(e, "");
+    } else {
+      this.handleDateValueChange(e, "");
+    }
   }
 
   render() {
-    let placeholderComponent = this.props.placeholderComponent ? (
-      this.props.placeholderComponent
-    ) : (
-      <TText tid="Placeholder" style={this.props.placeholderStyle}>
-        {this.props.placeholder}
-      </TText>
-    );
+    const { placeholderComponent, iconClear, iconRight, iconLeft } = this.props;
 
-    const valueString = this.props.dateTimeFormat(
-      this.state.date,
-      this.props.mode
-    );
+    const valueString = this.state.date
+      ? this.props.dateTimeFormat(this.state.date, this.props.mode)
+      : "";
 
     const timeValue = this.state.date || new Date();
+    const valueTestId = this.state.date
+      ? `Value[${this.state.date.getTime()}]`
+      : "Value[unknown]";
 
     return (
       <TestPathSegment name={`Field[${this.props.fieldRef}]` || "DatePicker"}>
         <View>
-          <Field
-            {...this.props}
-            ref="inputBox"
-            onPress={this._togglePicker.bind(this)}
-          >
+          <Field {...this.props} ref="inputBox" onPress={this._togglePicker}>
```

Comment:
Everytime I see a string ref, I'm reminded of the many time React have said they'll be removed.  Not for this PR though